source/tutorials/dev-environment.md

```diff
@@ -28,67 +13,104 @@ app = Flask(__name__)
 
 @app.route("/")
 def hello():
-    return "Hello, Nix!"
+    return {
+        "message": "Hello, Nix!"
+    }
 
 def run():
-    app.run(host="0.0.0.0")
+    app.run(host="0.0.0.0", port=5000)
 
 if __name__ == "__main__":
     run()
 ```
 
-And a `setup.py` script:
+This is a simple Flask application which serves a JSON document with the message
+"Hello, Nix!".
 
-```{code-block} python setup.py
-from setuptools import setup
+For the development environment, create a new file `shell.nix` which creates a
+shell environment to be used for development: 
 
-setup(
-    name='myapp',
-    version='0.1',
-    py_modules=['myapp'],
-    entry_points={
-        'console_scripts': ['myapp = myapp:run']
-    },
-)
-```
+```{code-block} nix shell.nix
+{ pkgs ? import <nixpkgs> {} }:
 
-Now build the package with:
+pkgs.mkShell {
+  packages = [
+    (pkgs.python3.withPackages (ps: [
+      ps.flask
+    ]))
 
-```{code-block} bash test_nix_build.sh
-nix-build
+    pkgs.curl
+    pkgs.jq
+  ];
+}
 ```
 
-This will create a symbolic link `result` to our package's path in the Nix store, which looks like `/nix/store/6i4l781jwk5vbia8as32637207kgkllj-myapp-0.1`. Look around to see what's inside.
+This creates a shell environment with a Python 3 that contains the `flask`
+library.
 
-You may notice we can run the application from the package like this: `./result/bin/myapp`. But we can also use the `default.nix` as a shell environment to get the same result:
+However, it also contains developer tools: `curl` (utility to perform web
+requests) and `jq` (utility to parse and format JSON documents). Both of them
+are not Python packages and they are not part of the Python ecosystem.
 
-```bash
-nix-shell default.nix
-python myapp.py
+If we went with Python's [virtualenv], it is not possible to have these
```

Comment:
Please move the reference link below this paragraph, it's hard to find in the source otherwise.